src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1726,6 +1732,13 @@
       <Kind Name="DiscardDesignation"/>
     </Field>
   </Node>
+  <Node Name="VarPatternSyntax" Base="PatternSyntax">
```

Comment:
Thought you were going to change DeclarationPatternSyntax to accept all designations. That's how DeclarationExpressionSyntax is defined.

Reply:
No, because the semantics of `var x` and `string x` are different, even if the input is of type `string`, moreover we do not intend to support anything like `string (x, y)`.

Reply:
Also, it would produce an ambiguity between a deconstruction pattern in which the type is specified (patterns between parens), and a var pattern with a parenthesized designator.

Reply:
How the shape of a SyntaxNode could cause an ambiguity?

I think we already parse it this way, except for the first step,

- If we encounter a `var`, we expect a **designation** and produce a **declaration pattern**
  e.g. `var _`, `var x`, `var (x, y)`

- otherwise, if the type is followed by an **identifier**, we produce a **declaration pattern**
  e.g. `T _`, `T id`

- otherwise, if the type is followed by `(` we expect a **deconstruction pattern**
  e.g. `Point ( subpattern, ...)`

- otherwise, if the type is followed by `{` we expect  a **property pattern**
  e.g. `Point { subpattern, ...}`

and so on.


>No, because the semantics of var x and string x are differen

We already represent both `var x` and `T x` with a `DeclarationPatternSyntax`, right?

Reply:
Now if the type is missing, we either have a deconstruction pattern or a property pattern based on the first token.

Reply:
> We already represent both `var x` and `T x` with a `DeclarationPatternSyntax`, right?

`var (x, y)` is what I will be calling a "var-pattern". Once things settle out, I expect patterns that starts with `var` to be parsed as a var-pattern.

Reply:
Right. I just suggested to use DeclarationPatternSyntax because (1) it would not be an API breaking change (for simple cases) (2) it has a duality with DeclarationExpressionSyntax which also accepts a parenthesized designation. Perhaps https://github.com/dotnet/roslyn/issues/23546 is the source of my confusion.